src/platform/Tizen/NetworkCommissioningThreadDriver.cpp

```diff
@@ -164,6 +164,36 @@ bool TizenThreadDriver::ThreadNetworkIterator::Next(Network & item)
     return false;
 }
 
+ThreadCapabilities TizenThreadDriver::GetSupportedThreadFeatures()
+{
+    BitMask<ThreadCapabilities> capabilites = 0;
+    if (CHIP_DEVICE_CONFIG_THREAD_BORDER_ROUTER)
```

Comment:
Is there a reason this has a generally different shape from the Linux/OpenThread bits?

Reply:
For some reason that I am not sure to understand Tizen was the only platform throwing me an error for the same code used in Linux/Openthread functions

```
INFO    ../../examples/chip-tool/third_party/connectedhomeip/src/platform/Tizen/NetworkCommissioningThreadDriver.cpp:171:109:   required from here
305
INFO    ../../examples/chip-tool/third_party/connectedhomeip/src/lib/support/BitMask.h:90:17: error: conversion from 'int' to 'chip::BitMask<chip::app::Clusters::NetworkCommissioning::ThreadCapabilitiesBitmap>::IntegerType' {aka 'short unsigned int'} may change value [-Werror=conversion]
306
INFO       90 |         updated |= static_cast<IntegerType>(bitMask & (value << shift));
307
INFO          |         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

When looking at the implementation of `constexpr BitMask & SetField(FlagsEnum mask, IntegerType value)` Everything seems cast correctly to `IntegerType`. So I do not quite get the error.The only thing I could think of is if Tizen architecture applies different types to defines vs enum but even then I don't grasp the error it throws as everything in the `SetField` is cast to IntegerType/ the passed value type.

I just ended up changing the implementation of this function so it doesn't use SetField for Tizen.

Reply:
I am guessing the issue is that `updated | static_cast<IntegerType>(bitMask & (value << shift))` gets promoted to `int` (which is actually standard-required behavior), and then the assignment back to `updated` fails?  I would have thought something on tizen already uses SetField before this PR, but maybe not...  This looks like a bug in BitMask.h that we should fix: it needs to cast to `IntegerType` after the bitwise `|`, not before.

Reply:
Oh! I didn't see it like that, thanks.
Updating BitMask with this and applying the same initial code to Tizen. We'll see the outcome thanks.